pkg/operator/resource/resourceapply/core.go

```diff
@@ -234,10 +234,18 @@ func ApplyConfigMap(client coreclientv1.ConfigMapsGetter, recorder events.Record
 }
 
 // ApplySecret merges objectmeta, requires data
-func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, required *corev1.Secret) (*corev1.Secret, bool, error) {
-	if len(required.StringData) > 0 {
-		return nil, false, fmt.Errorf("Secret.stringData is not supported")
+func ApplySecret(client coreclientv1.SecretsGetter, recorder events.Recorder, requiredInput *corev1.Secret) (*corev1.Secret, bool, error) {
+	// copy the stringData to data.  Error on a data content conflict inside required.  This is usually a bug.
+	required := requiredInput.DeepCopy()
```

Comment:
just curios why requiredInput was not checked for `nil` value here? It seems to be the case previously too.

Reply:
> just curios why `requiredInput` was not checked for `nil` value here? It is the same case before you made any changes too.

uh, hopefully it panics at some point.  don't do that.